core/common/src/main/java/alluxio/collections/ConcurrentHashSet.java

```diff
@@ -104,6 +104,9 @@ public boolean equals(Object o) {
       return true;
     }
     // allow equality with any Set.
+    if (!(o instanceof ConcurrentHashSet)) {
```

Comment:
It will work. It seems the common practice of defining equals is to require two candidates really have the identical class type.